aws/cloudformation/components/www_redirect.yml.erb

```diff
@@ -3,8 +3,59 @@
     # As CDO properties can vary between workstations, using them here is generally unsafe
     # Since this line is restricted to only run in production, we are comfortable using it.
     redirected_domains << "www.#{CDO.pegasus_hostname}" if rack_env?(:production)
+    
+    www_redirect_hosted_zone = Aws::Route53::Client.new.list_hosted_zones_by_name(dns_name: domain).hosted_zones.first
+    raise "Could not find #{domain} in hosted zones" unless www_redirect_hosted_zone.name.delete_suffix('.') == domain
+    www_redirect_hosted_zone_id = www_redirect_hosted_zone.id.delete_prefix("/hostedzone/")
+  -%>
+
+  <%=app%>DNS:
+    Type: AWS::Route53::RecordSetGroup
+    Properties:
+      HostedZoneId: <%=www_redirect_hosted_zone_id%>%
+      RecordSets:
+        <%- redirected_domains.each do |record_set_name| -%>
+        - Name: <%=record_set_name%>
+          Type: A
+          AliasTarget:
+            DNSName: !GetAtt [<%=app%>RedirectDistribution, DomainName]
+            HostedZoneId: Z2FDTNDATAQYW2 # static ID for CloudFront aliases
+        <%- end -%>
+
+  # Health checks and alarms for each DNS record we're handling 
+<%- redirected_domains.each do |domain_name| -%>
+  <%-
+    # Convert something like "autoscale-prod.code.org" to "AutoscaleProdCodeOrg"
+    logical_id = domain_name.titleize.gsub(/[^a-zA-Z0-9]/, "")
   -%>
   
+  <%=app%><%=logical_id%>HealthCheck:
+    Type: 'AWS::Route53::HealthCheck'
+    Properties: 
+      HealthCheckConfig: 
+        Type: HTTP
+        Port: 80
+        FailureThreshold: 3
+        FullyQualifiedDomainName: <%=domain_name%>
+        RequestInterval: 30
+
+
+  <%=app%><%=logical_id%>HealthCheckAlarm
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmDescription: "Health Check for <%=domain_name%>"
+      AlarmName: ""
+      Namespace: AWS/Route53
+      MetricName: HealthCheckStatus
+      ComparisonOperator: GreaterThanThreshold
+      Period: '60'
+      EvaluationPeriods: '3'
+      Threshold: '1'
+      Statistic: Average
+      AlarmActions:
+      - arn:aws:sns:us-east-1:475661607190:CDO-Urgent
```

Comment:
I think we want this Action conditional on whether this is the production environment.

Reply:
I've changed it to alarm to "CDO-Workday-Urgent" for nonprod. If it alarms, it needs to be looked at before attempting a DTP - but it doesn't need to wake anybody up.

```
alarm_topic = rack_env?(:production) ? "CDO-Urgent" : "CDO-WorkdayUrgent"
```